### ISSUE TITLE
Images don't print with #lang htdp/bsl

### ISSUE THREAD

**Issue by cena** [greghendershott/racket-mode#125]
I may be mistaken, but I thought [this comment on HackerNews](https://news.ycombinator.com/item?id=8346029) indicated that one could display images ala geiser in racket-mode.

Part of the problem may be that I may be assuming 'image' also applies to graphical shapes created by the likes of (circle ...) etc, but racket-mode doesn't seem to create them.

Caveat: new to racket, new to emacs, and this could simply be newbie confusion.  I'm trying to kill two birds with one stone by going through HtDP using emacs (vs DrRacket).  

This is what I'm getting when I attempt to draw a circle using the racket-mode REPL.  Geiser does display the circles correctly.

![rktmode](https://cloud.githubusercontent.com/assets/19334/7164583/b6bc717c-e366-11e4-96f0-8811867a1e61.png)

**Comment by greghendershott** [comment]
Interesting! Simplest possible example, this works for me with `#lang racket`

```
#lang racket
(require 2htdp/image)
(circle 10 "solid" "red")
```

But this does not, with `#lang htdp/bsl`:

```
#lang htdp/bsl
(require 2htdp/image)
(circle 10 "solid" "red")
```

I'll figure out why...

**Comment by cena** [comment]
I can't now remember why I wanted to use htdp/bsl - probably wanted to ensure that I was following the book/duplicating the DrR experience as closely as possible.

Using #lang racket does fix the image problem.  Using the same inputs as shown above, the difference now is that the output of (sqrt 2) doesn't have #i prepended (which may be a part of the htdp/bsl).  

Thanks!

**Comment by greghendershott** [comment]
I've spent quite a lot of time thinking about this, as well as trying out various alternative approaches. So far, no luck.

I also checked what Dr. Racket does. Like racket-mode, Dr. Racket also does _not_ print an image for `#lang htdp`, it prints `#(struct:object:image%...)`.

I'd be inclined to close this as "as-intended" or "won't-fix". Except you said it works in Geiser (I haven't yet tried that myself). Also, part of me wants to understand exactly _why_ it's not working. However I think I'll need to set this aside for awhile.

**Comment by cena** [comment]
Greg, thanks for taking the time to look into this.

What I find odd is that within DrRacket, if you specify (#lang htdp/bsl), you do get the #(struct:object...)

![screen shot 2015-04-20 at 10 00 10 am](https://cloud.githubusercontent.com/assets/19334/7233026/616c77e4-e744-11e4-9850-cc1aa5b1ef96.png)

However, if you choose the predefined 'Beginning Student' language (which I may be incorrectly assuming is the same as (#lang htdp/bsl)) through the GUI, it works correctly:

![screen shot 2015-04-20 at 10 01 04 am](https://cloud.githubusercontent.com/assets/19334/7233033/78272290-e744-11e4-9542-2ff939e380ad.png)

However, as you say - this doesn't appear to actually be a racket-mode issue, but rather an idiosyncrasy(bug?) with DrRacket.  @racket-lang did state that htdp/bsl support was experimental, so my apologies for taking up your time!

**Comment by greghendershott** [comment]
Thank you, that's an interesting and helpful clue. When I have time to dig in, I could look at what DrR does for the the pick-a-lang GUI, and, consider if that is something that racket-mode could and should do, too.

**Comment by Blaisorblade** [comment]
In my plt/racket#1024 issue, I linked to an earlier discussion.

TL;DR: [Felleisen declared 5 years ago "#lang htdp/bsl isn't ready for consumption."](http://narkive.com/H5koiJH7:3.313.43).

**Comment by cena** [comment]
Thanks, Blaisorblade - I did note that in my last post from 20 April (although my info came via tweet to @racket-lang.)

**Comment by Blaisorblade** [comment]
@cena Right, sorry for skimming too quickly! Now I even found your Twitter thread: https://twitter.com/racketlang/status/588060011856850945.

**Comment by greghendershott** [comment]
I'm going to close this, because realistically I don't think I'm going to be willing or able to get to this in the foreseeable future.

**Comment by greghendershott** [comment]
But, @cena this was a really nice quality bug report -- thank you for that, I really appreciated it!! 

I hope this is N/A for you because you've "graduated" to "real" `#lang racket`.

Thanks again.

**Comment by sorawee** [comment]
Today, someone asked the same question in Discord.

FYI: during my quick attempt to debug, I found that HtDP installs its own eval handler / display handler at [`htdp-lib/lang/private/sl-eval.rkt`](https://github.com/racket/htdp/blob/master/htdp-lib/lang/private/sl-eval.rkt) to deal with images. Not sure if it's relevant.
